daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala

```diff
@@ -738,10 +738,18 @@ private[lf] final class Compiler(
   private[this] def compileELet(elet: ELet) =
     withEnv { _ =>
       elet match {
-        case ELet(Binding(optBinder, _, bound), body) =>
-          let(withOptLabel(optBinder, compile(bound))) { boundPos =>
-            optBinder.foreach(addExprVar(_, boundPos))
-            compile(body)
+        case ELet(bindings, body) =>
+          val bounds = bindings.map { case Binding(optBinder, _, bound) =>
+            val se = withOptLabel(optBinder, compile(bound))
+            val p = nextPosition()
+            optBinder.foreach(addExprVar(_, p))
+            se
+          }
+          compile(body) match {
+            case SELet(bounds1, body1) =>
+              SELet(bounds ++ bounds1, body1)
```

Comment:
do we need to worry about the inner let having references to variables bound in outer let? I forgot if you can have references in multi-lets to previous bindings in daml-lf.